packages-exp/auth-exp/src/core/user/account_info.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as externs from '@firebase/auth-types-exp';
+
+import {
+  updateEmailPassword as apiUpdateEmailPassword,
+  UpdateEmailPasswordRequest
+} from '../../api/account_management/email_and_password';
+import { updateProfile as apiUpdateProfile } from '../../api/account_management/profile';
+import { User } from '../../model/user';
+import { _reloadWithoutSaving } from './reload';
+
+interface Profile {
+  displayName?: string | null;
+  photoURL?: string | null;
+}
+
+export async function updateProfile(
+  externUser: externs.User,
+  { displayName, photoURL: photoUrl }: Profile
+): Promise<void> {
+  if (displayName === undefined && photoUrl === undefined) {
```

Comment:
does this seem right? I can't clear both of them?

Reply:
Undefined doesn't clear; undefined just leaves the field be (no-op). `null` specifically clears. See the comment in the code example here: https://firebase.google.com/docs/reference/js/firebase.User#updateprofile